server/aws-lsp-codewhisperer/src/language-server/telemetryService.ts

```diff
@@ -199,4 +203,50 @@ export class TelemetryService extends CodeWhispererServiceToken {
             codeCoverageEvent: event,
         })
     }
+
+    public emitChatAddMessage(params: {
+        conversationId?: string
+        messageId?: string
+        customizationArn?: string
+        userIntent?: UserIntent
+        hasCodeSnippet?: boolean
+        programmingLanguage?: CodewhispererLanguage
+        activeEditorTotalCharacters?: number
+        timeToFirstChunkMilliseconds?: number
+        timeBetweenChunks?: number[]
+        fullResponselatency?: number
+        requestLength?: number
+        responseLength?: number
+        numberOfCodeBlocks?: number
+        hasProjectLevelContext?: number
+    }) {
+        if (!params.conversationId || !params.messageId) {
```

Comment:
should we make these params required?

Reply:
source where we get the conversationId and messageId from are optional. All the null/non-null checks i thought to better add here instead. 

One thing I can do here, when I move the respective event to destination in the telemetryService file, then I can see if we need to pass required from there then. Will take a note of this.